distributed/worker.py

```diff
@@ -1671,7 +1671,7 @@ async def close(
                         c._asynchronous = True
                         if c.asynchronous:
                             await c.close()
-                        else:
+                        else:  # pragma: no cover
```

Comment:
This entire section is very messy and I would love to get rid of it. I don't think we should ingore it
```suggestion
                        else:
```